### DISCUSSION TITLE
How to properly handle trailing slashes?

### DISCUSSION THREAD

**Discussion by zzzhangqi** [facebook/docusaurus#10726] [Q&A]
### Environment

Docusaurus Version: 3.6.0
Self-hosted: Nginx

Nginx.conf:
```conf
user  nginx;
worker_processes  2;
error_log  /var/log/nginx/error.log warn;
pid        /var/run/nginx.pid;
events {
    worker_connections  1024;
}
http {
    include       /etc/nginx/mime.types;
    default_type  application/octet-stream;
    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
                      '$status $body_bytes_sent "$http_referer" '
                      '"$http_user_agent" "$http_x_forwarded_for"';
    access_log  /var/log/nginx/access.log  main;
    sendfile        on;
    keepalive_timeout  65;
    client_header_buffer_size 16k;
    gzip  on;
    server {
        error_page  404   /404.html;
        listen       80;
        location / {
            root   /app;
            index  index.html index.htm;
            if ($http_accept_language !~* ^zh-CN){
                rewrite ^/$ /en/ redirect;
            }
        }
    }
}
```

### Question

When I click on the document in the sidebar, the URL does not carry `/`.

When I enter the `https://www.rainbond.com/docs/quick-start/quick-install` URL directly in the browser, it will automatically add `/` at the end, for example: `https://www.rainbond.com/docs/quick-start/quick-install/`

In `sitemap.xml`, `https://www.rainbond.com/docs/quick-start/quick-install` did not carry `/`. 

This would cause algolia crawler to not use sitemap to properly index my document.

Looking forward to reply.

**Comment by slorber** [comment]
We have a trailingSlash config, and your host Nging probably have one too. 

First you need to define if you want a trailing slash or not for the canonical/reference urls, and then configure both tools accordingly. Both are possible depending on hosts/tools you choose. 

Unfortunately it's specific to hosts/tools and we can't document all tools exhaustively.
Here's some basic doc I wrote: https://github.com/slorber/trailing-slash-guide

I don't know Nginx so unfortunately you'll have to figure this out yourself or ask the community on Discord. My intuition is that you should use either `true` or `false`, not `undefined`: 
https://docusaurus.io/docs/api/docusaurus-config#trailingSlash
